### PR TITLE
[NFC] Add a test case for user-defined tuple conformances.

### PR DESCRIPTION
**Pull request by hborla**
Checking this in to make sure it continues to work. Thanks to @slavapestov for the test case!

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

File: test/Generics/tuple-conformances.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-variadic-generics -parse-stdlib
+
+import Swift
+
+protocol P {
+ associatedtype A
```
Comment (slavapestov) [inline test/Generics/tuple-conformances.swift:6]: Something wonky with the formatting here (tabs vs spaces?)

**Review by slavapestov — APPROVED** [review]
Thanks!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test macOS

**Pull request merged**